[Inductor] Cache CUDA compilation errors

kadeng opened the pull request:
Summary: Add support for caching of CUDA (nvcc) compilation errors to codecache.py

Test Plan: CI ( for example Cutlass backend unit tests )

Reviewed By: ColinPeppler

Differential Revision: D71562040




cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @chenyang78 @muchulee8 @amjames @chauhang @aakhundov

ColinPeppler reviewed (approved):
reminder to apply lintrunner!

kadeng committed:
[Inductor] Cache CUDA compilation errors (#149716)

Summary:

Add support for caching of CUDA (nvcc) compilation errors to codecache.py

Test Plan: CI ( for example Cutlass backend unit tests )

Reviewed By: ColinPeppler

Differential Revision: D71562040

henrylhtsang commented:
@pytorchbot merge -i
